runtime/src/main/java/org/corfudb/runtime/collections/PersistedStreamingMap.java

```diff
@@ -159,7 +162,7 @@ public boolean containsValue(@NonNull Object value) {
     @Override
     public V get(@NonNull Object key) {
         getCounter.ifPresent(Counter::increment);
-        Optional<Timer.Sample> recordSample = MicroMeterUtils.startTimer(getCounter, SAMPLING_RATE);
+        Optional<Timer.Sample> recordSample = MicroMeterUtils.startTimer(getCounter, randomNumberGenerator.nextInt(SAMPLE_SPACE));
```

Comment:
This is a little convoluted, why is a counter needed? 

Can't you just do 

```java

if(sampling rate < PRNG.next(100)) {
sample
}
```

Reply:
Yes, this seems more straightforward. I had a different idea in mind.